lib/galaxy/tools/deps/mulled/get_tests.py

```diff
@@ -0,0 +1,304 @@
+    True
+    >>> test_search('snakemake:3.11.2--py34_1') == {'commands': ['snakemake --help > /dev/null'], 'imports': ['snakemake'], 'import_lang': 'python -c', 'container': 'snakemake:3.11.2--py34_1'}
+    True
+    >>> test_search('perl-yaml:1.15--pl5.22.0_0') == {'imports': ['YAML', 'YAML::Any', 'YAML::Dumper', 'YAML::Dumper::Base', 'YAML::Error', 'YAML::Loader', 'YAML::Loader::Base', 'YAML::Marshall', 'YAML::Node', 'YAML::Tag', 'YAML::Types'], 'import_lang': 'perl -e', 'container': 'perl-yaml:1.15--pl5.22.0_0'}
+    True
+    """
+    if deep:  # do a deep search
+        return deep_test_search(container, recipes_path, anaconda_channel, github_repo)
+    # else shallow
+    result = try_a_func(get_test_from_anaconda, get_anaconda_url, (container, anaconda_channel), container)
+    if result:
+        return result
+    return {'container': container}
+
+
+def hashed_test_search(container, recipes_path=None, deep=False, anaconda_channel='bioconda', github_repo='bioconda/bioconda-recipes'):
+    """
+    Get test for hashed containers
+    >>> t = hashed_test_search('mulled-v2-0560a8046fc82aa4338588eca29ff18edab2c5aa:c17ce694dd57ab0ac1a2b86bb214e65fedef760e-0')
+    >>> t == {'commands': ['bamtools --help'], 'imports': [], 'container': 'mulled-v2-0560a8046fc82aa4338588eca29ff18edab2c5aa:c17ce694dd57ab0ac1a2b86bb214e65fedef760e-0', 'import_lang': 'python -c'}
+    True
+    """
+    package_tests = {'commands': [], 'imports': [], 'container': container, 'import_lang': 'python -c'}
+
+    githubpage = requests.get('https://raw.githubusercontent.com/BioContainers/multi-package-containers/master/combinations/%s.tsv' % container)
+    if githubpage.status_code == 200:
+        packages = githubpage.text.split(',')  # get names of packages from github
+        packages = [package.split('=') for package in packages]
+    else:
+        packages = []
+
+    containers = []
+    for package in packages:
+        r = requests.get("https://anaconda.org/bioconda/%s/files" % package[0])
+        p = '-'.join(package)
+        for line in r.text.split('\n'):
+            if p in line:
+                build = line.split(p)[1].split('.tar.bz2')[0]
+                if build == "":
+                    containers.append('%s:%s' % (package[0], package[1]))
+                else:
+                    containers.append('%s:%s-%s' %
+                                      (package[0], package[1], build))
+                break
+
+    for container in containers:
+        tests = test_search(container, recipes_path, deep, anaconda_channel, github_repo)
+        package_tests['commands'] += tests.get('commands', [])  # not a very nice solution but probably the simplest
+        for imp in tests.get('imports', []):
+            package_tests['imports'].append("%s 'import %s'" % (tests['import_lang'], imp))
+
+    return package_tests
+
+
+doctest.testmod()
```

Comment:
I'd suggest creating real tests instead of doctests. We're currently using pytest, but you can also use unittest tests if you like.